### DISCUSSION TITLE
How to simulate no cache

### DISCUSSION THREAD

**Discussion by gits00** [mortbopet/Ripes#215] [Q&A]
Hi, 
I am new to Ripes. Can the simulator be used with no cache configured? I need to see the effect on execution time with a particular cache config and one with no cache at all.
Thanks.

**Comment by mortbopet** [comment]
Hi @gits00 

When you say execution time, what do you exactly mean? The time that it takes for your own computer to simulate a program, or the simulated execution time of a program (# instructions retired relative to # of cycles executed)?

If the latter (which i assume), then it's important to recognize that the Ripes cache simulator does _not_ simulate memory access latency. And by extension, the simulated execution time of the program is not affected by the cache configuration.  
The cache simulator is (for now) only meant to simulate the interactions between a program and a given cache configuration.

**Comment by gits00** [comment]
Thanks for clarifying.

Yes, I am looking for # of instructions retired or cycles of a program running on Ripes. 

So, can I safely assume that when I run my program all the memory accesses don't cause a stall, and runtime is purely affected by the CPU architecture?

**Comment by mortbopet** [comment]
> So, can I safely assume that when I run my program all the memory accesses don't cause a stall, and runtime is purely affected by the CPU architecture?

Yep, but i would suggest that you try it out for yourself just to be sure of this. That is, simulate a program and try different cache configurations. The execution stats shown on the processor tab should be independent of your cache config.
